pallets/connectors-gateway/connectors-gateway-routers/src/axelar_evm.rs

```diff
@@ -0,0 +1,191 @@
+#[derive(Debug, Encode, Decode, Clone, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+pub struct AxelarEVMRouter<T>
+where
+	T: frame_system::Config
+		+ pallet_connectors_gateway::Config
+		+ pallet_ethereum_transaction::Config
+		+ pallet_evm::Config,
+{
+	pub router: EVMRouter<T>,
+	pub evm_chain: EVMChain,
+	pub connectors_contract_address: H160,
+	pub _marker: PhantomData<T>,
+}
+
+impl<T> AxelarEVMRouter<T>
+where
+	T: frame_system::Config
+		+ pallet_connectors_gateway::Config
+		+ pallet_ethereum_transaction::Config
+		+ pallet_evm::Config,
+	T::AccountId: AsRef<[u8; 32]>,
+{
+	/// Calls the init function on the EVM router.
+	pub fn do_init(&self) -> DispatchResult {
+		self.router.do_init()
+	}
+
+	/// Encodes the Connectors message to the required format,
+	/// then executes the EVM call using the generic EVM router.
+	pub fn do_send(&self, sender: AccountIdOf<T>, msg: MessageOf<T>) -> DispatchResult {
+		let eth_msg = get_axelar_encoded_msg(
+			msg.serialize(),
+			self.evm_chain.to_string(),
+			self.connectors_contract_address.to_string(),
+		)
+		.map_err(DispatchError::Other)?;
+
+		self.router.do_send(sender, eth_msg)
+	}
+}
+
+/// Encodes the provided message into the format required for submitting it
+/// to the Axelar contract which in turn submits it to the Connectors
+/// contract.
+pub(crate) fn get_axelar_encoded_msg(
+	serialized_msg: Vec<u8>,
+	target_chain: String,
+	target_contract: String,
+) -> Result<Vec<u8>, &'static str> {
+	// Centrifuge -> `callContract` on the Axelar Gateway contract.
+	//
+	// Axelar Gateway contract -> `handle` on the Connectors gateway contract
+	// deployed on Ethereum.
+
+	// Connectors Call:
+	//
+	// function handle(bytes memory _message) external onlyRouter {}
+
+	#[allow(deprecated)]
+	let encoded_connectors_contract = Contract {
+		constructor: None,
+		functions: BTreeMap::<String, Vec<Function>>::from([(
+			CONNECTORS_FUNCTION_NAME.to_string(),
+			vec![Function {
+				name: CONNECTORS_FUNCTION_NAME.into(),
+				inputs: vec![Param {
+					name: CONNECTORS_MESSAGE_PARAM.into(),
+					kind: ParamType::Bytes,
+					internal_type: None,
+				}],
+				outputs: vec![],
+				constant: false,
+				state_mutability: Default::default(),
+			}],
+		)]),
+		events: Default::default(),
+		errors: Default::default(),
+		receive: false,
+		fallback: false,
+	}
+	.function(CONNECTORS_FUNCTION_NAME)
+	.map_err(|_| "cannot retrieve Connectors contract function")?
+	.encode_input(&[Token::Bytes(serialized_msg)])
+	.map_err(|_| "cannot encode input for Connectors contract function")?;
+
+	// Axelar Call:
+	//
+	// function callContract(
+	//     string calldata destinationChain,
+	//     string calldata destinationContractAddress,
+	//     bytes calldata payload,
+	// ) external {
+	//     emit ContractCall(
+	// 			msg.sender,
+	// 			destinationChain,
+	// 			destinationContractAddress,
+	// 			keccak256(payload),
+	// 			payload,
+	// 	   );
+	// }
```

Comment:
Was this here to aid your Rust coding and or do you still want to keep it?

Reply:
Yeah sounds good. Maybe leave it as a more obvious comment and less like forgotten commented code or leave a link to the actual definition of that type/call in the original repo?

Reply:
OK, updated the function comment with links to both contracts that are handled here.